src/coreclr/vm/eventing/eventpipe/ds-rt-coreclr.h

```diff
@@ -329,6 +330,39 @@ ds_rt_disable_perfmap (void)
 #endif // FEATURE_PERFMAP
 }
 
+static
+uint32_t
+ds_rt_appcontext_properties_get (dn_vector_ptr_t *props_array)
+{
+	STATIC_CONTRACT_NOTHROW;
+	EP_ASSERT (props_array != NULL);
+
+	return Configuration::EnumerateKnobs([](const LPCWSTR& name, const LPCWSTR& value, void* context) {
```

Comment:
I've tapped into the `Configuration` class for this enumeration since it is the owner of the app config properties starting [here](https://github.com/dotnet/runtime/blob/main/src/coreclr/dlls/mscoree/exports.cpp#L312) and the diagnostic server/command handling is started later via [here](https://github.com/dotnet/runtime/blob/main/src/coreclr/dlls/mscoree/exports.cpp#L320).

The other statically available places to access this information were possibly from the host or coreclr exports and contracts, but I felt that it was not appropriate to add the enumeration API there for external consumption; also, based on discussion in #83756, there may be some desire to filter the properties before the `AppContext` gets to report them, which feels like it would naturally fit in at configuration initialization. Finally, I will have a follow up PR that allows modification of these values, which the Configuration class can be more easily made to do than say the host and coreclr properties since those are already "sealed" at this point in time.